### PR TITLE
Add F.softmax / nn.Softmax support to torch_nntile

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Implements `torch.nn.functional.softmax` and `nn.Softmax` on the `nntile` (PrivateUse1) device by registering PyTorch ATen ops `_softmax`, `_softmax.out`, and `_softmax_backward_data`.

## Changes

- **Forward:** `tensor::maxsumexp` + `tensor::softmax` via new `tensor_softmax_fp32` executor function
- **Backward:** Composed from `tensor::sumprod_slice`, `tensor::add_slice`, and `tensor::multiply_inplace` (same formula as `NNSoftmaxOp` in libnntile)
- **Refactor:** Generalized `maxsumexp_graph_shape` to accept an arbitrary axis (cross-entropy call sites updated)
- **New files:** `nntile_softmax.cpp`, `nntile_softmax_backward.cpp`
- **Tests:** `test_softmax_parity.py` — forward/backward parity for `dim=-1`, `dim=0`, and `nn.Softmax`

## Limitations (v1)

- `float32` contiguous tensors only
- `half_to_float=True` not supported
- `F.log_softmax` not included (follow-up)

## Testing

```bash
pytest -vv torch_nntile/tests/test_softmax_parity.py
```

All 5 parity tests pass on CPU-only build with libnntile.

<div><a href="https://cursor.com/agents/bc-85be4cf9-3c1e-4817-9564-770f86f2f99e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-85be4cf9-3c1e-4817-9564-770f86f2f99e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review
